docs/assignment3.md

```diff
@@ -1 +1,306 @@
+
+
+`State diagram over Game Class`
+![](imgs/ass3%20stateDiagram%20Game.png)
+
+The state diagram above represents the state a Game can be in. It has three different properties that can be changes and accessed with different methods.   
+
+
+## Sequence diagrams									
+Author(s): `Teona Moga`, `Patricia Santana Vasquez`, `Noe Rousset`
+`Sequence diagram showing League creation and execution`
+![](imgs/teamcreation.png)
+
+This sequence diagram represent how the creation of the player's team occur and the interactions between the game and team editor.
+The player represent by the actor element provides a list of players for his team in the team editor. Once the "Save Team" button is pressed, createTeam() is called.
+
+The team editor checks if the player selection is correct, as shown in the Alt fragment. If there are not enough players to create a team, showError() is called and displays an error message showing which fields are empty.
+If the same player appears multiple times in the selection, showError() is called and an error message is displayed. If an error occurs, createTeam() exits and the player must press "Save Team" to start again.
+
+If there are no errors, addTeam() is called, and :Game creates a new team object and adds it to the team pool.
+
+`Sequence diagram showing League creation and execution`
+![](imgs/Sequence2League.png)
+
+This sequence diagrams shows the interactions between the player and the game to execute the league from a player provided schedule. This sequence happens when all teams have been made and the player starts the league simulation. 
+The player represented by the actor element provides a schedule with includes the number of matches and which teams will play in the first round.
+
+Once the Start button is pressed, the league simulator checks for errors as shown in the alt fragment. If there are matches missing teams, showError is called and an error message is displayed. This causes the league execution to stop and the player has to hit the start button again.
+ 
+If there are no errors, the League object is created and the execution of the league begins.
+The league starts with the playRound function. Here all the scheduled matches are played one by one in the Loop fragment. A match object is created and returns the winner of the match before being destroyed. The winning teams are added to the pool of teams playing next round. 
+
+Once a round is over, the Alt fragment checks if there are more than one remaining team. If yes playRound is called recursively, as shown by the self-call element. The exit condition is one team remaining, and when it is met the results are sent to the game and the league object is destroyed.
+The full results are then displayed by LeagueSimulatorPane.
+
+
+## Implementation									
```

Comment:
Besides the inconsistency in the inclusion of comments and the rather large number of very similar statements (utilize iterators !), The code is rather clean and concise.
**[CLARIFICATION]** if you however notice that your implementation grade has gone down in the rubric, it is due to the missing F4 and the incomplete justification of changes you made to F1